kie-maven-plugin/src/main/java/org/kie/maven/plugin/BuildMojo.java

```diff
@@ -180,27 +175,60 @@ private void buildDrl() throws MojoFailureException {
             ResultsImpl messages = (ResultsImpl)kieBuilder.getResults();
 
             List<Message> errors = messages != null ? messages.filterMessages( Message.Level.ERROR): Collections.emptyList();
+
+            if (container != null) {
+                Map<String, Object> kieMap = getKieMap();
+                if (!kieMap.isEmpty()) {
+                    CompilerHelper helper = new CompilerHelper();
+                    String compilationID = helper.getCompilationID(kieMap, getLog());
+                    shareKieObjects(kModule,
```

Comment:
the use of internal `share*` methods and the use of helper.share* is confuse here. Why not improve and do something like:
helper.shareData(this)

Reply:
Ok, was an attempt to preserve the infos with methodNames about the object shared